hapi/base.py

```diff
@@ -97,7 +100,10 @@ def _create_request(self, conn, method, url, headers, data):
         return params
 
     def _gunzip_body(self, body):
-        sio = StringIO.StringIO(body)
+        if body and isinstance(body, bytes):
```

Comment:
Isn't this just six.BytesIO?

Reply:
According to link http://python-future.org/compatible_idioms.html#stringio here, there are two options: io.BytesIO and io.StringIO for different strings.

Reply:
Given that you added `six` in order to get python2/3 compatibility: https://six.readthedocs.io/#six.BytesIO is the one class that you need.

Reply:
Ah, actually, ```io.BytesIO``` is also available in python 2. I think I could also just use ```io.BytesIO``` here in order to be consistent with other usages of ```io.BytesIO``` in this library?

Reply:
Does this method get called with a `str` or `unicode` or sometimes one, sometimes the other? Maybe you need to use `six.BytesIO`

Reply:
If that is the case, then in python 3, it will also be called with ```bytes``` or ```str```. In this case, ```six.BytesIO``` will also not work since it is just an alias for ```io.BytesIO``` in python 3.

So to be safe, maybe we still need to retain the check here. Check whether it is bytes or unicode?